homestar-runtime/tests/network.rs

```diff
@@ -0,0 +1,183 @@
+use crate::utils::{kill_homestar, retrieve_output, stop_homestar, BIN_NAME};
+use anyhow::Result;
+use once_cell::sync::Lazy;
+use predicates::prelude::*;
+use serial_test::file_serial;
+use std::{
+    path::PathBuf,
+    process::{Command, Stdio},
+    thread,
+    time::Duration,
+};
+
+#[allow(dead_code)]
+static BIN: Lazy<PathBuf> = Lazy::new(|| assert_cmd::cargo::cargo_bin(BIN_NAME));
+
+#[test]
+#[file_serial]
+fn test_libp2p_generates_peer_id_serial() -> Result<()> {
+    let _ = stop_homestar();
+
+    let homestar_proc = Command::new(BIN.as_os_str())
+        .arg("start")
+        .arg("-c")
+        .arg("tests/test_node1/config/settings.toml")
+        .arg("--db")
+        .arg("homestar1.db")
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let dead_proc = kill_homestar(homestar_proc);
+    let stdout = retrieve_output(dead_proc);
+
+    assert_eq!(
+        true,
+        predicate::str::contains("local peer ID generated").eval(stdout.as_str())
+    );
+    assert_eq!(
+        true,
+        predicate::str::contains("peer_id=12D3KooWBYAug7e9eE7z1z1jaYjfVQCfRy8r3keVYD8jdsEGZHr")
+            .eval(stdout.as_str())
+    );
+
+    Ok(())
+}
+
+#[test]
+#[file_serial]
+fn test_libp2p_listens_on_address_serial() -> Result<()> {
+    let _ = stop_homestar();
+
+    let homestar_proc = Command::new(BIN.as_os_str())
+        .arg("start")
+        .arg("-c")
+        .arg("tests/test_node1/config/settings.toml")
+        .arg("--db")
+        .arg("homestar1.db")
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let dead_proc = kill_homestar(homestar_proc);
+    let stdout = retrieve_output(dead_proc);
+
+    assert_eq!(
```

Comment:
don't need eq, just assert! on the predicate

Reply:
Yep! Changed this on all the tests. 👍